docs/api.md

```diff
@@ -50,8 +50,10 @@ mainly, but not only, provisioning details.
 
 * *image* -- Holds details for the image to be deployed on a given host.
   * *url* -- The URL of an image to deploy to the host.
-  * *checksum* -- The actual checksum or a URL to a file containing
-    the checksum for the image at *image.url*.
+  * *checksum* -- The actual md5 checksum or a URL to a file containing
+    the md5 checksum for the image at *image.url*. Other checksum algorithms
+    can be specified using the format: hash_algo:hash_value
+    e.g. `sha256:6623803...`, `sha256:https://...`, etc
```

Comment:
Can we add a list of the allowed algorithms? Is it just md5 and sha256?

Reply:
In other places we have tried to hiding behavior behind URL query arguments. In this case, the change would make the value into something that isn't a valid URL. Is there a reason to not simply add a new field to the API to hold the checksum algorithms? That would let us validate the checksum value and would make it easy to declare the allowed values using an enum.

Reply:
It was suggested earlier by @hardys to re-use the existing API. Adding a new field would make things simpler and we don't have to worry about parsing the checkusm field.

Reply:
Since adding a dedicated filed under image, this point is now moot.

Reply:
I've added an unit test and have restricted the hash algos.